#!/usr/bin/env python
PACKAGE = "tbf_gripper_perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("cv_icp_iterations",    int_t,    0, "Number of ICP iterations",100, 0, 10000)
gen.add("cv_icp_tolerance", 	double_t, 0, "Controls the accuracy of registration at each iteration of ICP.", 		   .005, 0,     1)
gen.add("cv_icp_rejectionScale",double_t, 0, "Robust outlier rejection is applied for robustness. This value actually corresponds to the standard deviation coefficient. Points with rejectionScale * &sigma are ignored during registration.",  	2.5, 0,   100)
gen.add("cv_icp_numLevels",    	int_t,    0, "Number of pyramid levels to proceed. Deep pyramids increase speed but decrease accuracy. Too coarse pyramids might have computational overhead on top of the inaccurate registrtaion. This parameter should be chosen to optimize a balance. Typical values range from 4 to 10.",  	  8, 0,   20)

exit(gen.generate(PACKAGE, "tbf_gripper_perception", "IcpMatching"))